sum_types_generator/lib/src/sum_type_spec.dart

```diff
@@ -163,3 +186,5 @@ String _resolveTypeName(
 
   return _resolveTypeName(type);
 }
+
+enum ParameterStyle { positional, named, empty }
```

Comment:
I think you can "default" to `positional` in the no-payload case. "Empty" doesn't seem to fit naturally into `ParameterStyle` type, because, given a parameter, it's either positional or named and cannot be empty. Of course, I understand, why you may want to have `empty` here, but there are other ways to handle such cases and, this time, I'd suggest to just default to `positional`. This should not confuse anybody (figuratively speaking), because looking at `parameterStyle` and `parameterName` properties only makes sense, when the payload is required (so, that should be checked first).

And I'd suggest moving this enum definition just below `CaseSpec` class. In this file, it seems, first all types are defined, then some functions.

Reply:
Sure, make sense to me.